server/routes/offenceCodeDecisions/offenceCodeDecisions.ts

```diff
@@ -4,72 +4,90 @@ import decisionTree from '../../offenceCodeDecisions/DecisionTree'
-
+    const { errors } = pageData
+    const { user } = res.locals
+    const decisionForm = pageData // The form backing this page
+    const placeholderValues = await this.placeholderValues(adjudicationNumber, user)
+    const decision = decisionTree.findByUrl(req.path.replace(`/${adjudicationNumber}/${incidentRole}/`, ''))
     const pageTitle = decision.getTitle().getProcessedText(placeholderValues, incidentRole as IncidentRole)
     const questions = decision.getChildren().map(d => {
       return {
         id: d.id(),
         label: d.getQuestion().getProcessedText(placeholderValues),
+        type: d.getType().toString(),
       }
     })
-    return res.render(`pages/${decision.getPage() || 'offenceCodeDecisions'}`, {
-      errors: error ? [error] : [],
-      selectedDecisionId,
+    const selectedDecisionViewData = await this.viewDataFromDecisionForm(decisionForm, user)
+    return res.render(`pages/offenceCodeDecisions`, {
+      errors: errors || [],
+      decisionForm,
+      selectedDecisionViewData,
       questions,
       pageTitle,
       pageData,
     })
   }
 
-  view = async (req: Request, res: Response): Promise<void> => this.renderView(req, res, {})
+  view = async (req: Request, res: Response): Promise<void> => {
+    if (req.query.selectedPerson) {
+      // We are coming back from a user selection. We want to record this in the DecisionForm stored on the session and
+      // then redirect to the view page after removing the request parameter.
+      updateSessionDecisionForm(req, req.query.selectedPerson as string)
+      return res.redirect(
+        url.format({
+          pathname: this.urlHere(req),
+        })
+      )
+    }
+    // We are viewing this page. If we have come from a user selection then we should have the previous state of the
+    // form in the session, so we render that now and remove it from the session.
+    return this.renderView(req, res, getAndDeleteSessionDecisionForm(req) || {})
+  }
 
   submit = async (req: Request, res: Response): Promise<void> => {
     const { adjudicationNumber, incidentRole } = req.params
-    const { selectedDecisionId } = req.body
-
-    const error = validateForm({ selectedDecisionId })
+    const decisionForm = decisionFormFromPost(req)
+    const searching = !!req.body.searchUser // Is this a standard submit or are we searching for a user.
+    const errors = validateForm(decisionForm, searching)
 
-    if (error) {
+    if (errors) {
       return this.renderView(req, res, {
-        error,
-        selectedDecisionId,
+        errors,
+        ...decisionForm,
       })
     }
 
-    const selectedDecision = decisionTree.findById(selectedDecisionId)
+    const selectedDecision = decisionTree.findById(decisionForm.selectedDecisionId)
+
+    if (searching) {
+      // We need to redirect the user to the search page, but before we do that we need to save the current data from
+      // the session.
+      setSessionDecisionForm(req, decisionForm)
```

Comment:
This is the only time we create a decision form on the session. 

So in overview. 
1) We create decision form on the session when we are redirecting to the search
2) We update that decision form from the request parameter when coming back from a search and redirect to the view
3) We read the  decision form from the session in the view and delete it.
Those should be the only interactions with the session.